crates/gpui/src/platform/mac/text_system.rs

```diff
@@ -93,26 +93,16 @@ impl PlatformTextSystem for MacTextSystem {
     }
 
     fn all_font_names(&self) -> Vec<String> {
-        let collection = core_text::font_collection::create_for_all_families();
-        let Some(descriptors) = collection.get_descriptors() else {
-            return Vec::new();
-        };
-        let mut names = BTreeSet::new();
-        for descriptor in descriptors.into_iter() {
-            names.extend(lenient_font_attributes::family_name(&descriptor));
+        let mut names = Vec::new();
+        let lock = self.0.read();
+        if let Ok(fonts_in_system) = lock.system_source.all_families() {
+            names.extend(fonts_in_system);
         }
-        if let Ok(fonts_in_memory) = self.0.read().memory_source.all_families() {
+        if let Ok(fonts_in_memory) = lock.memory_source.all_families() {
             names.extend(fonts_in_memory);
         }
-        names.into_iter().collect()
-    }
-
-    fn all_font_families(&self) -> Vec<String> {
-        self.0
-            .read()
-            .system_source
-            .all_families()
-            .expect("core text should never return an error")
+        drop(lock);
```

Comment:
Do you need to drop the lock explicitly? It's going out of scope on the next line anyways.

Reply:
Sorry, I forgot to delete this.